packages/cli/openapi-parser/src/getExtension.ts

```diff
@@ -30,3 +31,20 @@ export function getExtension<T>(
     }
     return undefined;
 }
+
+export function getExtensionAndValidate<T>(
+    object: object,
+    extension: Extension<T>,
+    schema: z.ZodSchema,
+    context: AbstractOpenAPIV3ParserContext
+): T | undefined {
+    try {
+        const extensionValue = getExtension<T>(object, extension);
+        if (extensionValue != null) {
+            return schema.parse(extensionValue);
+        }
+    } catch (e) {
+        context.logger.error(`Failed to parse ${extension}`);
```

Comment:
can this function take a set of breadcrumbs or keys so that you can log: 

```
`${breadcrumbs.join("->")}: Failed to parse extension ${extension}`
```

Reply:
Then atleast the error message will read: 

```sh
> [error]: POST /abc/123: Failed to parse extension `x-codeSamples`
```